EIPS/eip-5496.md

```diff
@@ -0,0 +1,207 @@
+    event PrivilegeTotalChanged(uint256 newTotal, uint256 oldTotal);
+
+    /// @notice set the privilege holder of a NFT.
+    /// @dev expires should better be less than 30 days
+    /// Throws if `msg.sender` is not approved or owner of the tokenId.
+    /// @param tokenId The NFT to set privilege for
+    /// @param privilegeId The privilege to set
+    /// @param user The privilege holder to set
+    /// @param expires For how long the privilege holder can have
+    function setPrivilege(uint256 tokenId, uint256 privilegeId, address user, uint256 expires) external;    
+
+    /// @notice Check if a privilege has expired
+    /// @param tokenId The identifier of the queried NFT
+    /// @param privilegeId The identifier of the queried privilege
+    /// @return Whether a user has a certain privilege
+    function privilegeExpires(uint256 tokenId, uint256 privilegeId) external view returns(uint256);
+
+    /// @notice Check if a user has a certain privilege
+    /// @param tokenId The identifier of the queried NFT
+    /// @param privilegeId The identifier of the queried privilege
+    /// @param user The address of the queried user
+    /// @return Whether a user has a certain privilege
+    function hasPrivilege(uint256 tokenId, uint256 privilegeId, address user) external view returns(bool);
+}
+```
+Every contract implementing the `ERC721P` SHOULD set a maximum privilege number before setting any privilege, the privilegeId MUST NOT be greater than the maximum privilege number.
+
+The `PrivilegeAssigned` event MUST be emitted when `setPrivilege` is called.
+
+The `PrivilegeTransfered` event MUST be emitted when `transferPrivilege` is called.
+
+The `supportsInterface` method MUST return `true` when called with `0x076e1bbb`.
+
+
+```solidity
+/// @title ERC-721P Cloneable extension - Optional
+interface IERC721PCloneable {
+    /// @notice Emitted when set the `privilege ` of a NFT cloneable.
+    event PrivilegeCloned(uint tokenId, uint privId, address from, address to);
+
+    /// @notice set a certain privilege cloneable
+    /// @param tokenId The identifier of the queried NFT
+    /// @param privilegeId The identifier of the queried privilege
+    /// @param referrer The address of the referrer
+    /// @return Whether the operation is sucessful or not
+    function clonePrivilege(uint tokenId, uint privId, address referrer) external returns (bool);
+}
+```
+The `PrivilegeCloned` event MUST be emitted when `clonePrivilege` is called.
+
+## Rationale
+
+### Multiple privileges
+
+Current ERC-721 standard only record the ownership and its transfer, the privileges of an NFT are not recorded on-chain. This extension would allow merchants/projects to give out a certain privilege to a specified group of people, and owners of the privileges can manage each one of the privileges independently. This facilitates a great possibility for NFTs to have real usefulness. 

```

Comment:
```suggestion
EIP-721 only records the ownership and its transfer, the privileges of an NFT are not recorded on-chain. This extension would allow merchants/projects to give out a certain privilege to a specified group of people, and owners of the privileges can manage each one of the privileges independently. This facilitates a great possibility for NFTs to have real usefulness. 
```
Required: Replace ERC with EIP
Required: Fix grammar
Recommended: rewording